operational_analysis/methods/plant_analysis.py

```diff
@@ -634,9 +641,60 @@ def process_reanalysis_data(self):
             (None)
         """
 
-        # Define empty data frame that spans past our period of interest
+        # Identify start and end dates for long-term correction
+        # First find date range common to all reanalysis products and drop minute field of start date
+        start_date = max(
+            [self._plant._reanalysis._product[key].df.index.min() for key in self._reanal_products]
+        ).replace(minute=0)
+        end_date = min(
+            [self._plant._reanalysis._product[key].df.index.max() for key in self._reanal_products]
+        )
+
+        # Next, update the start date to make sure it corresponds to a full time period
+        start_date_minus = start_date - pd.DateOffset(hours=1)
+        if (self.time_resolution == "M") & (start_date.month == start_date_minus.month):
+            # If not at the beginning of a month, use the beginning of the next month as the start date
+            start_date = start_date.replace(day=1, hour=0, minute=0) + pd.DateOffset(months=1)
+        elif (self.time_resolution == "D") & (start_date.day == start_date_minus.day):
+            # If not at the beginning of a day, use the beginning of the next day as the start date
+            start_date = start_date.replace(hour=0, minute=0) + pd.DateOffset(days=1)
+
+        # Now determine the end date based on either the user-defined end date or the end of the last full month
+        if self.end_date_lt is not None:
```

Comment:
The end date logic also took me a little bit to digest, and was resistant to it at first, but again, I ended up liking it.

Reply:
After going through the tests as well, I like this whole start/end check even more. Nice work!

Reply:
Thanks Rob, always open to suggestions if you see a clearer way of writing something.

Reply:
That was the thing, is I was trying to think of a different way, but I couldn't come up with anything more succinct/clear/etc., so I still think leaving it as-is is the right thing to do.